src/server/pfs/server/testing/server_v2_test.go

```diff
@@ -619,31 +624,84 @@ func (v *validator) recordFileSet(files fileSetSpec) {
 	}
 }
 
-func (v *validator) validate(r io.Reader) error {
-	hdr, err := tar.NewReader(r).Next()
-	if err != nil {
-		// We expect an empty tar stream if no files were uploaded.
-		if err == io.EOF && len(v.files) == 0 {
-			return nil
+func (v *validator) validate(t *testing.T, r io.Reader) {
+	expected := v.files.makeTarStream()
+	diffTarStreams(t, expected, r)
+	// _, err := tar.NewReader(r).Next()
+	// if err != nil {
+	// 	// We expect an empty tar stream if no files were uploaded.
+	// 	if err == io.EOF && len(v.files) == 0 {
+	// 		return nil
+	// 	}
+	// 	return err
+	// }
+
+	// var filesSorted []string
+	// for file := range v.files {
+	// 	filesSorted = append(filesSorted, file)
+	// }
+	// sort.Strings(filesSorted)
+	// for _, file := range filesSorted {
+	// 	buf := &bytes.Buffer{}
+	// 	if _, err := io.CopyN(buf, r, int64(len(v.files[file]))); err != nil {
+	// 		fmt.Println("missing", file)
+	// 		require.NoError(err)
+	// 	}
+	// 	// if !bytes.Equal(v.files[file], buf.Bytes()) {
+	// 	// 	fmt.Println("VALIDATE", len(v.files[file]), buf.Len())
+	// 	// 	return errors.Errorf("file %v's header and/or content is incorrect", file)
+	// 	// }
+	// }
+}
+
+func diffTarStreams(t *testing.T, a, b io.Reader) error {
```

Comment:
This doesn't seem to check the content. I think the approach we want to take is what originally existed, which basically just compares the serialized content. We also will want to add some configuration for the larger workloads to check hashes rather than storing and checking actual content.